baybe/utils/basic.py

```diff
@@ -121,31 +123,56 @@ def to_tuple(x: Sequence) -> tuple:
     return tuple(x)
 
 
-def filter_attributes(
+def match_attributes(
     object: Any,
     callable_: Callable,
-    ignore: Collection[str] = ("self", "kwargs", "args"),
-) -> dict[str, Any]:
+    /,
+    strict: bool = True,
+    ignore: Collection[str] = (),
```

Comment:
@AVHopp, @Scienfitz, FYI: Had to add one commit to extend the logic here, because `qNIPV` carries some attributes that are not passed down to the `gpytorch` class. Thumbs up / resolve if happy

Reply:
discussion already ongoing in other thread, hence closing here